File: lecture_6/mario2/mario.py
# A more advanced version of mario.py

from cs50 import get_int

def main():
    while True:
        n = get_int("Please input a number between 1 and 8, incusive: ")
        if 1 <= n <= 8:
            break

    for i in range(n):
        print(" " * (n - 1 - i), end="")
        print("#" * (i + 1), end="")
        print(" " * 2, end="")
        print("#" * (i + 1), end="")
        print()

main()

File: lecture_1/pset1/mario2/mario.c
#include <stdio.h>
#include <cs50.h>

// Declare the functions used in the main program
void print(char c, int n);

int main(void)
{
    int n;
    do
    {
        n = get_int("Please enter a number between the numbers 1 and 8: ");
    }
    while (n < 1 || n > 8);

    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            if (i + j >= n - 1)
                printf("#");
            else
                printf(" ");
        }

        // The shape is divided by two spaces
        print(' ', 2);
        // Print the second half of the shape, which we know is i + 1
        print('#', i + 1);
        // Move to the next line / loop
        printf("\n");
    }
}

// Create a print function to print to make the second half of the shape easier
void print(char c, int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("%c", c);
    }
}